src/main/java/makaroshyna/onlinebookstore/service/cartitem/CartItemServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package makaroshyna.onlinebookstore.service.cartitem;
+
+import jakarta.transaction.Transactional;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import makaroshyna.onlinebookstore.dto.cartitem.CreateCartItemRequestDto;
+import makaroshyna.onlinebookstore.dto.cartitem.UpdateCartItemRequestDto;
+import makaroshyna.onlinebookstore.exception.EntityNotFoundException;
+import makaroshyna.onlinebookstore.mapper.CartItemMapper;
+import makaroshyna.onlinebookstore.model.CartItem;
+import makaroshyna.onlinebookstore.model.ShoppingCart;
+import makaroshyna.onlinebookstore.repository.cartitem.CartItemRepository;
+import makaroshyna.onlinebookstore.service.book.BookService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final BookService bookService;
+
+    @Override
+    @Transactional
+    public CartItem save(CreateCartItemRequestDto requestDto, ShoppingCart shoppingCart) {
+        CartItem cartItem = cartItemMapper.toModel(requestDto);
+        cartItem.setShoppingCart(shoppingCart);
+        String title = bookService.getById(requestDto.getBookId()).getTitle();
+        cartItem.getBook().setTitle(title);
+
+        return cartItemRepository.save(cartItem);
+    }
+
+    @Override
+    @Transactional
+    public CartItem update(Long id, UpdateCartItemRequestDto requestDto) {
+        Optional<CartItem> optionalCartItem = cartItemRepository.findById(id);
+
+        if (optionalCartItem.isPresent()) {
+            CartItem cartItem = optionalCartItem.get();
+            cartItem.setQuantity(requestDto.quantity());
+            return cartItemRepository.save(cartItem);
+        }
+
+        throw new EntityNotFoundException("Cannot find cart item with id " + id);
```

Comment:
Let's stick to the [fail early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)
```suggestion
        CartItem cartItem = cartItemRepository.findById(id).orElseThrow(...);        
        cartItem.setQuantity(requestDto.quantity());
        return cartItemRepository.save(cartItem);
```